backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/service/impl/BPartnerDAO.java

```diff
@@ -567,7 +567,14 @@ public CountryId getDefaultShipToLocationCountryIdOrNull(final BPartnerId bpartn
 	public CountryId getBPartnerLocationCountryId(@NonNull final BPartnerLocationId bpartnerLocationId)
 	{
 		final I_C_BPartner_Location bpLocation = getBPartnerLocationByIdEvenInactive(bpartnerLocationId);
-		return CountryId.ofRepoId(bpLocation.getC_Location().getC_Country_ID());
+
+		if(bpLocation != null)
+		{
+			return CountryId.ofRepoId(bpLocation.getC_Location().getC_Country_ID());
+		}
+
+		// if it was not found outOfTrx, try to get it in trx
+		return retrieveBPartnerLocationCountryIdInTrx(bpartnerLocationId);
```

Comment:
it should really be found out-of-trx...isn't it?

Reply:
Right. This was not even needed. My country location not being found here was caused by those locations not being loaded correctly from the builder. Got rid of this check.